test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs

```diff
@@ -1443,6 +1468,55 @@ public async Task GetTrustResultAsync_WithTamperedRepositoryCountersignedPackage
                     }
                 }
             }
+
+            [CIOnlyFact]
+            public async Task GetTrustResultAsync_WithExpiredPrimaryCertificateAndExpiredRepositoryCertificateAndNoTimestamps_ReturnsValidAsync()
+            {
+                TimestampService timestampService = await _fixture.GetDefaultTrustedTimestampServiceAsync();
+
+                using (X509Certificate2 authorCertificate = await GetExpiringCertificateAsync(_fixture))
+                using (X509Certificate2 repoCertificate = await GetExpiringCertificateAsync(_fixture))
+                using (Test test = await Test.CreateAuthorSignedRepositoryCountersignedPackageAsync(
+                    authorCertificate,
+                    repoCertificate,
+                    timestampService.Url,
+                    timestampService.Url))
+                using (var packageReader = new PackageArchiveReader(test.PackageFile.FullName))
+                {
+                    await SignatureTestUtility.WaitForCertificateExpirationAsync(authorCertificate);
+                    await SignatureTestUtility.WaitForCertificateExpirationAsync(repoCertificate);
+
+                    PrimarySignature primarySignature = await packageReader.GetPrimarySignatureAsync(CancellationToken.None);
+                    PackageVerificationResult status = await _provider.GetTrustResultAsync(packageReader, primarySignature, _defaultSettings, CancellationToken.None);
+
+                    Assert.Equal(SignatureVerificationStatus.Valid, status.Trust);
```

Comment:
Do we still trust if both timestamp and repro cert are expired?

Reply:
In accept mode, which is the default, yes.